src/Tizen.Peripheral/Tizen.Peripheral/I2c.cs

```diff
@@ -0,0 +1,151 @@
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using NativeI2c = Interop.Peripheral.I2c;
+
+namespace Tizen.Peripheral
+{
+    /// <summary>
+    /// The class allows applications to communicate via i2c platform's bus.
+    /// </summary>
+    /// <privilege>http://tizen.org/privilege/peripheralio</privilege>
+    public class I2c : IDisposable
+    {
+        /// <summary>
+        /// Native handle to I2c.
+        /// </summary>
+        protected IntPtr _handle;
+
+        private I2c(IntPtr handle)
+        {
+            _handle = handle;
+        }
+
+        /// <summary>
+        /// Opens an I2C slave device.
+        /// </summary>
+        /// <param name="bus">The I2C bus number that the slave device is connected.</param>
+        /// <param name="address">The address of the slave device.</param>
+        /// <returns>The I2c object.</returns>
+        public static I2c Open(int bus, int address)
+        {
+            var ret = NativeI2c.Open(bus, address, out IntPtr handle);
+            if (ret != Internals.Errors.ErrorCode.None)
+                throw ExceptionFactory.CreateException(ret);
+
+            return new I2c(handle);
+        }
+
+        /// <summary>
+        /// Closes an I2C slave device.
+        /// </summary>
+        public void Close()
+        {
+            var ret = NativeI2c.Close(_handle);
+            if (ret != Internals.Errors.ErrorCode.None)
+                throw ExceptionFactory.CreateException(ret);
+        }
+
+        /// <summary>
+        /// Reads the bytes data from the I2C slave device.
+        /// </summary>
+        /// <param name="size">The size of array to read.</param>
+        /// <returns>The array of bytes.</returns>
+        public byte[] Read(int size)
+        {
+            var length = Convert.ToUInt32(size);
+            byte[] data = new byte[size];
+            var ret = NativeI2c.Read(_handle, data, length);
+            if (ret != Internals.Errors.ErrorCode.None)
+                throw ExceptionFactory.CreateException(ret);
+
+            return data;
+        }
+
+        /// <summary>
+        /// Writes the bytes data to the I2C slave device.
+        /// </summary>
+        /// <param name="data"></param>
+        public void Write(byte[] data)
+        {
+            var length = Convert.ToUInt32(data.Length);
+            var ret = NativeI2c.Write(_handle, data, length);
+            if (ret != Internals.Errors.ErrorCode.None)
+                throw ExceptionFactory.CreateException(ret);
+        }
+
+        /// <summary>
+        /// Reads single byte data from the register of the I2C slave device.
+        /// </summary>
+        /// <param name="register">The register address of the I2C slave device to read.</param>
+        /// <returns>The single byte data.</returns>
+        public byte ReadRegisterByte(byte register)
```

Comment:
Handling Exceptions is more expensive than simply writing if statements.
How about making a method in the form of `public bool TryReadRegisterByte(byte register, out byte value)` so that even when a read fails, it can be handled with one if statement rather than an exception handling?

Other Read and Write functions will all be similar.
As I know that reading/writing to a pin frequently fails depending on the situation.

Reply:
I would have to redesign whole API.